librz/main/rz-find.c

```diff
@@ -498,6 +498,14 @@ static int rzfind_open(RzfindOptions *ro, const char *file) {
 		free(buf);
 		return res;
 	}
+	if (ro->exec_command) {
+		char *command = rz_str_newf("%s %s", ro->exec_command, file);
+		int status = rz_sys_system(command);
+		if (status == -1) {
+			RZ_LOG_ERROR("Failed to execute command: %s", command);
+		}
+		free(command);
+	}
```

Comment:
no. the function that actually does the search is `rzfind_open_file`. please try to understand what the request is and what the function does. the command needs to be executed only when there is a hit with the search. this means that each time the program finds something that matches what it is looking for it should either print (current behaviour) or execute a program (new feature you are implementing).

Also you have removed the previous command `-E` which was used for ESIL search (even tho there was no help message info on the util) and didn't remove all the old usages.

Reply:
It's clear now, will fix it. thanks